theMind.js

```diff
@@ -64,11 +67,10 @@ var calculateShortestPath = function([startY, startX], castle) {
 };
 
 var locationStatus = function(position, castle) {
-  var castleSize = castle.length;
   var dft = position.distanceFromTop;
   var dfl = position.distanceFromLeft;
 
-  if(dfl < 0 || dfl >= castleSize || dft < 0 || dft >= castleSize) {
+  if(dfl < 0 || dfl >= 10 || dft < 0 || dft >= 3) {
```

Comment:
Changed search limits from 10 up to down, and 10 right toleft, to 3 up todown, and 10 right to left.